### PR TITLE
Fixed codacy issue related to security and duplication

### PR DESCRIPTION
**Pull request by gaurkrgaurav**
Signed-off-by: Gaurav Gaur <gaurav.gaur@seagate.com>

# Problem Statement
- Fixed Codacy issues related to security and optimized the code.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by gaurkrgaurav** [comment]
@kanchan-chaudhari @ravindrachoudhari  Please review it.

File: scripts/beck/st/error_injection.py [hunk 1]
```diff
@@ -167,27 +171,35 @@ def ReadTypeSize(byte):  # Ex: 0001(ver) 0009(type) 00003dd8(size)
...
         wbfr.write(b'\x33\x33\x44\x44\x22\x22\x11\x11')
         wbfr.seek(offset)
         ReadMetadata(offset)
 
 
-# If you want to verify the written Metadata then run below segment of code
 def ReadMetadata(offset):
+    """
+    If you want to verify the written Metadata then run below segment of code
+    """
```
Comment (nikitadanilov) [inline scripts/beck/st/error_injection.py]: "Verifies that meta-data contains the valid footer at the given offset."

File: scripts/beck/st/error_injection.py [hunk 2]
```diff
@@ -200,7 +212,11 @@ def ReadCompleteRecord(offset):
     curr_record = [ hex(int(i, 16)) for i in curr_record]
     return curr_record, offset # Return record data and footer offset
 
+
 def ReadBeBNode(offset):
+    """
+    Read BeNode data
+    """
```
Comment (nikitadanilov) [inline scripts/beck/st/error_injection.py]: Function documentation is in 3rd person, see doc/coding-style.md. That is "Reads", "Induces", etc.

**Review by nikitadanilov — CHANGES_REQUESTED** [review]

File: scripts/beck/st/error_injection.py [hunk 3]
```diff
@@ -20,10 +20,10 @@
 import binascii
 import sys
 import os
-import random
 import argparse
 import time
 import logging
+import random
```
[the remark this answers is not in the document]
Comment (gaurkrgaurav) [reply scripts/beck/st/error_injection.py:26]: Actually, I was trying some other library, After some experiments come back to the same one. Just a Typo

File: scripts/beck/st/error_injection.py (hunk 1 above)
Comment (gaurkrgaurav) [reply scripts/beck/st/error_injection.py]: Acknowledge

**Review by ketanarlulkar — APPROVED** [review]

File: scripts/beck/st/error_injection.py [hunk 4]
```diff
@@ -167,27 +171,32 @@ def ReadTypeSize(byte):  # Ex: 0001(ver) 0009(type) 00003dd8(size)
...
-        else:
-            return False, data
+        return False, data
+
 
-# This function returns complete data with list format in 8 bytes chunks,
-# starting after header and until footer for record
 def ReadCompleteRecord(offset):
+    '''
+
```
Comment (kanchan-chaudhari) [inline scripts/beck/st/error_injection.py:196]: Remove Extra line

File: scripts/beck/st/error_injection.py [hunk 5]
```diff
@@ -266,83 +283,40 @@ def InduceRandomCorruption(noOfErr):
...
+    for recType in recList:
         logger.info("Record Name: {}".format(recType))
         lookupList = recordDict[recType]
         length = len(lookupList)
         if length == 0:
             logger.info("Record List is empty. Moving to Next Record")
         else:
-            lookupList = random.sample(lookupList, random.randint(1, length))
+            lookupList = random_system.sample(lookupList,
+                                                    random_system.randint(1, length))
```
Comment (kanchan-chaudhari) [inline scripts/beck/st/error_injection.py:299]: Alignment.

File: scripts/beck/st/error_injection.py [hunk 6]
```diff
@@ -380,8 +354,13 @@ def Induce512kbError():
                     count = count + 1
     return count
 
-# This function take stob_f_container, stob_f_key and returns cob_f_cotainer, cob_f_key and device_id
+
 def ConvertAdstob2Cob(stob_f_container, stob_f_key):
+    '''
+
```
Comment (kanchan-chaudhari) [inline scripts/beck/st/error_injection.py:360]: Remove extra line

**Review by ravindrachoudhari — APPROVED** [review]

File: scripts/beck/st/error_injection.py (hunk 6 above)
Comment (gaurkrgaurav) [reply scripts/beck/st/error_injection.py:360]: This check was given by codacy:
1 blank line required between summary line and description (found 0) (D205)

**Review by kanchan-chaudhari — APPROVED** [review]

**Comment by mehjoshi** [comment]
retest this please

**Comment by gaurkrgaurav** [comment]
@nikitadanilov can you please review it again?

**Review by mehjoshi — APPROVED** [review]

**Pull request merged**